portfolio/src/main/java/com/google/sps/servlets/PageVisitsProcessor.java

```diff
@@ -0,0 +1,181 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.gson.Gson;
+
+import javax.servlet.http.HttpServletRequest;
+import java.time.LocalDate;
+import java.time.ZoneOffset;
+import java.util.Calendar;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+public class PageVisitsProcessor {
+
+  private static final int MAX_DAYS = 30;
+  private static final String VISITS_TODAY = "Visits Today";
+
+  // Define constants for "Visits Today" entity properties
+  private static final String VISITS = "visits";
+  private static final String DATE = "date";
+  private static final String DAY_OF_WEEK = "dayOfWeek";
+  private static final String TIMESTAMP = "timestamp";
+
+  // Define constants for chart options (an enum might be a better choice)
```

Comment:
I agree with your comment about an enum being a better choice.

Reply:
Is there a good way to convert an integer value into an enum? I brainstormed ways to do this, but none of the methods I came up with seemed efficient.

Reply:
Maybe you could use enum fields/method as described on [this page](http://tutorials.jenkov.com/java/enums.html)?

It's been a while since I've coded in Java, so feel free to ignore this if you think this way is better.